### PR TITLE
resolve docker image tags so vulnerability scanning always works

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/compute

### CODE REVIEW

File: plugins/gcloud/lib/samson_gcloud/tag_resolver.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    class << self
+      def resolve_docker_image_tag(image)
+        return unless SamsonGcloud.gcr? image
+        return if image.match? Build::DIGEST_REGEX
+
+        tag = image.split(":", 2)[1] || DEFAULT_TAG
+
+        success, json = Samson::CommandExecutor.execute(
+          "gcloud", "container", "images", "list-tags", image, "--limit", LIMIT.to_s, "--format", "json",
```
Comment (adammw) [inline plugins/gcloud/lib/samson_gcloud/tag_resolver.rb]: can't you use the container registry api directly rather than using something gcr specific?

Comment (adammw) [reply plugins/gcloud/lib/samson_gcloud/tag_resolver.rb]: https://docs.docker.com/registry/spec/api/#listing-image-tags

Comment (adammw) [reply plugins/gcloud/lib/samson_gcloud/tag_resolver.rb]: (that being said, using it against gcr requires docker auth credentials - not sure if you have them in samson or not already)

Comment (grosser) [reply plugins/gcloud/lib/samson_gcloud/tag_resolver.rb]: it works, is easy to debug locally, reuses the gcloud executable we already use in different places, does not need new credentials ... so lots of upsides :)

File: plugins/gcloud/lib/samson_gcloud/tag_resolver.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    class << self
+      def resolve_docker_image_tag(image)
+        return unless SamsonGcloud.gcr? image
+        return if image.match? Build::DIGEST_REGEX
+
+        image, tag = image.split(":", 2)
+        tag ||= DEFAULT_TAG
+
+        success, json = Samson::CommandExecutor.execute(
+          "gcloud", "container", "images", "list-tags", image, "--limit", LIMIT.to_s, "--format", "json",
```
Comment (adammw) [inline plugins/gcloud/lib/samson_gcloud/tag_resolver.rb]: just use `gcloud container images describe <image_name>:<tag>` -> take `fully_qualified_digest`?

Comment (grosser) [reply plugins/gcloud/lib/samson_gcloud/tag_resolver.rb]: done

**Review by rbayerl — APPROVED** [review]

**Pull request merged**